pkg/scheduler/generic_scheduler_test.go

```diff
@@ -1068,13 +1069,16 @@ func TestFindFitAllError(t *testing.T) {
 		t.Errorf("unexpected error: %v", err)
 	}
 
-	// TODO(#94696): use cmp.Diff() to compare `diagnosis`.
-	if len(diagnosis.NodeToStatusMap) != len(nodes) {
-		t.Errorf("unexpected failed status map: %v", diagnosis.NodeToStatusMap)
+	expected := framework.Diagnosis{
+		NodeToStatusMap: framework.NodeToStatusMap{
+			"1": framework.NewStatus(framework.Unschedulable, st.ErrReasonFake).WithFailedPlugin("MatchFilter"),
+			"2": framework.NewStatus(framework.Unschedulable, st.ErrReasonFake).WithFailedPlugin("MatchFilter"),
+			"3": framework.NewStatus(framework.Unschedulable, st.ErrReasonFake).WithFailedPlugin("MatchFilter"),
```

Comment:
I have updated the failed plugin name manually, the test still pass. So I doubt that `cmp.Diff` is really compare each field one by one.

Reply:
I tried to modify the Status.Equal function, and now your problem can be solved. But I am not sure about the consequences of modifying Status.Equal...